src/inputs/SelectField.stories.tsx

```diff
@@ -197,6 +198,61 @@ export const Contrast = Template.bind({});
 // @ts-ignore
 Contrast.args = { compact: true, contrast: true };
 
+const loadTestOptions: TestOption[] = zeroTo(1000).map((i) => ({ id: String(i), name: `Project ${i}` }));
+
+export function PerfTest() {
+  const [selectedValue, setSelectedValue] = useState<string>(loadTestOptions[2].id);
+  return (
+    <SelectField
+      label="Project"
+      value={selectedValue}
+      onSelect={setSelectedValue}
+      errorMsg={selectedValue !== undefined ? "" : "Select an option. Plus more error text to force it to wrap."}
+      options={{
+        initial: [loadTestOptions[2]],
+        load: async () => {
+          return new Promise((resolve) => {
+            // @ts-ignore - believes `options` should be of type `never[]`
+            setTimeout(() => resolve({ options: loadTestOptions }), 1500);
+          });
+        },
+      }}
+      onBlur={action("onBlur")}
+      onFocus={action("onFocus")}
+    />
+  );
+}
+PerfTest.parameters = { chromatic: { disableSnapshot: true } };
+
+export function LoadingState() {
+  const [selectedValue, setSelectedValue] = useState<string>(loadTestOptions[2].id);
+
+  return (
+    <SelectField
+      label="Project"
+      value={selectedValue}
+      onSelect={setSelectedValue}
+      options={{
+        initial: [loadTestOptions[2]],
+        load: async () => {
+          return new Promise((resolve) => {
+            // @ts-ignore - believes `options` should be of type `never[]`
+            setTimeout(() => resolve({ options: loadTestOptions }), 5000);
+          });
+        },
+      }}
+    />
+  );
+}
+LoadingState.decorators = [withDimensions()];
+LoadingState.parameters = {
+  chromatic: { delay: 1000 },
+};
+LoadingState.play = async ({ canvasElement }: { canvasElement: HTMLElement }) => {
```

Comment:
**suggestion:** It would be helpful if you could add a comment on what this does! 

Just checked the the Storybook and see that there are alignment weirdness 

<img width="678" alt="CleanShot 2022-01-10 at 14 08 23@2x" src="https://user-images.githubusercontent.com/3606121/148816815-0f0d901b-2dee-48ee-9f7b-45c07fb8fd36.png">

Reply:
Yea, the alignment is off a bit when the field is flush on the edge of the screen. Not sure why, I think its a React-Aria thing, but haven't dug into it.